remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -132,6 +135,34 @@ public void addItem(RemixerItem remixerItem) {
     return list;
   }
 
+  /**
+   * Notifies all other instances of variables with the same key that the value has changed and
+   * informs the synchronization mechanism as well.
+   */
+  void onValueChanged(Variable variable) {
+    synchronizationMechanism.onValueChanged(variable);
+    List<RemixerItem> itemList = getItemsWithKey(variable.getKey());
+    for (RemixerItem item : itemList) {
+      if (item != variable) {
+        ((Variable) item).setValueWithoutNotifyingOthers(variable.getSelectedValue());
```

Comment:
For a while now, this thought has been creeping in my mind. Maybe having multiple SynchronizationMechanism that the client provides is not the right pattern. 

An alternative pattern is for Remixer to use a single implementation of SynchronizationMechanism, but allow that SynchronizationMechanism to be configured by the client. Eg:

```
class Remixer {
  private final SynchronizationMechanism sync = new RemixerSynchronizationMechanism();

  public SynchronizationMechanism getSynchronizationMechanism() {
    return this.sync;
  }
}
```
```
class ClientApplication extends Application {
  void onCreate() {
    super.onCreate();

    Remixer.getInstance().getSynchronizationMechanism().sharedPreferencesSupport = true;
    Remixer.getInstance().getSynchronizationMechanism().firebaseSupport = false;
  }
}
```

This alternative proposal would get rid of a lot of the "icky feelings" I have about the current pattern (those feelings are hard to organize into words).

A huge advantage to this is exactly what you said in your message. It would allow the SynchronizationMechanism to own more of this logic that I'd like to see pulled out of Remixer. Right now there's a little bit of syncing logic in Remixer and a little bit in SynchronizationMechanism. I'd like to see all of it go to SynchronizationMechanism.

This would include:
- When a new variable is added, the SynchronizationMechanism checks all others to make sure the new variable is legal.
- When a variable value changes, the SynchronizationMechanism updates all other equivalent variables.